indico/modules/events/registration/controllers/display.py

```diff
@@ -110,20 +111,25 @@ def _process(self):
                        ~RegistrationForm.is_deleted,
                        ~Registration.is_deleted,
                        _join=Registration.registration_form)
-                 .order_by(db.func.lower(Registration.last_name), db.func.lower(Registration.first_name)))
-        registrations = [(reg.get_full_name(), reg.get_personal_data()) for reg in query]
+                 .order_by(*Registration.order_by_name)
+                 .options(contains_eager('registration_form')))
+
+        registrations = [(reg.get_full_name(), reg.get_personal_data(), reg) for reg in query]
```

Comment:
What about already including the checked-in status in the third element of that tuple, i.e. `reg.checked_in and reg.registration_form.publish_checkin_enabled`? That would also make the `show_checkin_status` logic easier (`any(x[2] for x in registrations)`)